lib/rubocop/cop/root_cops/factories/factory_name.rb

```diff
@@ -22,6 +27,8 @@ def on_send(node)
 
             factory_symbol_name = *node.arguments[0]
             factory_name_array = factory_symbol_name[0].id2name.split("__")
+            factory_name_array.shift if factory_name_array.size > 1 && factory_name_array[0] == @system_name
```

Comment:
Made a minor change here, to add the `size > 1` guard which was failing for a policy factory in the policy system. ❌

Reply:
Can we add a spec for that?

Reply:
So added. See the spec for `"/code/server/systems/policy/spec/factories/policies.rb"`.